### PR TITLE
Add a basic healthcheck url

### PR DESCRIPTION
**Pull request by lgarvey**

### CODE REVIEW

File: requirements-dev.in [hunk 1]
```diff
@@ -8,6 +8,7 @@ pytest-env
 pytest-cov
 factory-boy
 freezegun
+mock
```
Comment (marcofucci) [inline requirements-dev.in]: you don't need this ;-)

File: config/urls.py [hunk 2]
```diff
@@ -20,5 +19,6 @@
     url(r'^access-denied/$', TemplateView.as_view(template_name='sso/access-denied.html'), name='access-denied'),
     url(r'^email/', include('sso.emailauth.urls')),
     url(r'^', include('sso.localauth.urls', namespace='localauth')),
-]
 
+    url(r'^check$', HealthCheckView.as_view(), name='healthcheck')
```
Comment (marcofucci) [inline config/urls.py]: trailing slash? I think all the urls have that

File: sso/emailauth/views.py [hunk 3]
```diff
@@ -74,4 +74,5 @@ def get_context_data(self, **kwargs):
 
         context['next_url'] = self.request.GET.get('next', '')
 
-        return context
\ No newline at end of file
+        return context
+
```
Comment (marcofucci) [inline sso/emailauth/views.py]: extra blank line

File: sso/healthcheck/middleware.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        request.start_time = time.time()
+
+        response = self.get_response(request)
+
+        return response
+
```
Comment (marcofucci) [inline sso/healthcheck/middleware.py]: extra blank line

File: sso/healthcheck/views.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['status'] = 'OK' if self._do_check() else 'FAIL'
+
+        # nearest approximation of a response time
+        context['response_time'] = time.time() - self.request.start_time
+
+        return context
+
```
Comment (marcofucci) [inline sso/healthcheck/views.py]: extra blank line to remove

File: sso/tests/test_healthcheck.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+        response = client.get('/check')
+
+        assert response.status_code == 200
+
+        xml = ET.fromstring(response.content)
+        status = xml[0].text
+
+        assert status == 'FAIL'
+
```
Comment (marcofucci) [inline sso/tests/test_healthcheck.py]: extra blank line to remove

File: sso/templates/sso/health-check.html [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<pingdom_http_custom_check>
+    <status>{{ status }}</status>
+    <response_time>{{ response_time }}</response_time>
+</pingdom_http_custom_check>
```
Comment (marcofucci) [inline sso/templates/sso/health-check.html]: no blank line, can you add one :)

File: sso/healthcheck/views.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import time
+
+from django.views.generic import TemplateView
+
+from sso.user.models import User
```
Comment (marcofucci) [inline sso/healthcheck/views.py]: if would be cleaner to use `django.contrib.auth.get_user_model` instead

File: sso/healthcheck/views.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class HealthCheckView(TemplateView):
+    template_name = 'sso/health-check.html'
+
+    def _do_check(self):
+        """Perform a basic DB test"""
+        try:
+            User.objects.all().count()
```
Comment (marcofucci) [inline sso/healthcheck/views.py]: `.exists()` should be more efficient than `.count()`

File: sso/healthcheck/views.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+            return True
+        except Exception:
```
Comment (marcofucci) [inline sso/healthcheck/views.py]: Don't we want to log the original exception and maybe send it to sentry or something?

File: config/urls.py (hunk 2 above)
Comment (lgarvey) [reply config/urls.py]: Yeh, Mat specifically showed an example that didn't have it.  However, let's follow the django convention.

File: requirements-dev.in (hunk 1 above)
Comment (lgarvey) [reply requirements-dev.in]: Yes, doh! It's already in the unittest module.

File: sso/healthcheck/views.py (hunk 10 above)
Comment (lgarvey) [reply sso/healthcheck/views.py]: Maybe, if the database does die, sentry will start freaking out anyway.

File: sso/tests/test_healthcheck.py [hunk 11]
```diff
@@ -13,7 +13,7 @@
 class TestHealthCheck:
     def test_check_view(self, client):
 
-        response = client.get('/check')
+        response = client.get('/check/')
```
Comment (marcofucci) [inline sso/tests/test_healthcheck.py]: Can't you use resolve? https://docs.djangoproject.com/en/1.11/ref/urlresolvers/#resolve

Comment (lgarvey) [reply sso/tests/test_healthcheck.py]: Yep! Let's resolve ...

Comment (lgarvey) [reply sso/tests/test_healthcheck.py]: You mean reverse() surely?

Comment (marcofucci) [reply sso/tests/test_healthcheck.py]: oh yeah sorry, multitasking :/

**Pull request merged**